master/internal/provisioner/aws_spot.go

```diff
@@ -186,11 +195,16 @@ func (c *awsCluster) listSpot(ctx *actor.Context) ([]*Instance, error) {
 	}
 
 	if canceledButInstanceRunningReqs.numReqs() > 0 {
-		ctx.Log().Infof(
+		ctx.Log().Debugf(
 			"terminating EC2 instances associated with canceled spot requests: %s",
 			strings.Join(canceledButInstanceRunningReqs.idsAsList(), ","),
 		)
-		c.terminateOnDemand(ctx, canceledButInstanceRunningReqs.idsAsListOfPointers())
+		_, err = c.terminateInstances(canceledButInstanceRunningReqs.instanceIds())
```

Comment:
question: why do you intentionally not emit any logs for those terminated instances?

Reply:
This is the confusing logging I mention in the description. This is background cleanup and showing the logs leads to very confusing log messages where the same instance keeps getting terminated.

Reply:
IIUC, we terminate all the instances associated with the requests having the status code `request-canceled-and-instance-running`. Why not just terminate such instances that are running so we don't get spamming logs.

Reply:
It requires a second round trip to AWS and doesn't really provide any value.